### PR TITLE
Provide support for JSON output in `tblite` flavour

### PR DESCRIPTION
**Pull request by marcelmbn**
### JSON output for `multicharge`

- add `json_results` subroutine (and associated service routines) from `tblite`  with following properties
   - atomic charges
   - summed electrostatic energy
   - gradient of energy vs. cartesian coordinates
- add boolean for JSON output
- add `validation` tests to `multicharge`

### CODE REVIEW

**Comment by awvwgk** [comment]
There is a CI failure, can you have a look whether this is related to this change?

**Comment by marcelmbn** [comment]
> There is a CI failure, can you have a look whether this is related to this change?

Apparently, the `intel oneAPI` cannot be set up correctly (from `CI / build (ubuntu-latest, meson, debug, intel, 2021) (pull_request)`:
```
:: initializing oneAPI environment ...
   aeb9d2f7-4873-4062-b64d-c0777c6bb885.sh: BASH_VERSION = 5.1.16(1)-release
   args: Using "$@" for setvars.sh arguments: 
:: compiler -- latest
:: debugger -- latest
:: dev-utilities -- latest
:: mkl -- latest
:: mpi -- latest
:: tbb -- latest
:: oneAPI environment initialized ::
 
/home/runner/work/_temp/aeb9d2f7-[48](https://github.com/grimme-lab/multicharge/actions/runs/10165719713/job/28147107002?pr=29#step:12:49)73-4062-b64d-c0777c6bb885.sh: line 2: /opt/intel/oneapi/compiler/2024.0/env/vars.sh: No such file or directory
Error: Process completed with exit code 1.
##[debug]Finishing: Setup Intel oneAPI environment
```

For `macos-latest` (also not related to the PR):
```
Run actions/setup-python@v1
Error: Version 3.8 with arch x64 not found
Available versions:
```

**Comment by awvwgk** [comment]
Can you open an issue? This needs to be fixed before any other patch can be merged.

**Comment by marcelmbn** [comment]
CI fix has been merged.

**Comment by marvinfriede** [comment]
What about a simple test?

**Comment by marcelmbn** [comment]
> What about a simple test?

Done.

**Comment by awvwgk** [comment]
The new argument should be documented in https://github.com/grimme-lab/multicharge/blob/main/man/multicharge.1.adoc as well.

File: app/main.f90 [hunk 1]
```diff
@@ -201,8 +216,10 @@ subroutine get_arguments(input, input_format, grad, charge, error)
             call fatal_error(error, "Invalid charge value")
             exit
          end if
-      case("-grad", "--grad")
+      case("-g", "--grad")
```
Comment (awvwgk) [inline app/main.f90:219]: This change is unrelated
```suggestion
      case("-grad", "--grad")
```

**Review by awvwgk — APPROVED** [review]
